internal/hcp/registry/hcl.go

```diff
@@ -70,6 +79,27 @@ func (h *HCLMetadataRegistry) CompleteBuild(
 	return h.bucket.completeBuild(ctx, name, artifacts, buildErr)
 }
 
+// IterationRemainingBuilds returns the message to print in the UI if the iteration is not yet done
+func (h *HCLMetadataRegistry) IterationStatusSummary(ui sdkpacker.Ui) {
+	rem := h.bucket.Iteration.RemainingBuilds()
+	if rem == nil {
+		return
+	}
+
+	buf := &strings.Builder{}
+
+	buf.WriteString(fmt.Sprintf(
+		"Iteration %q is not complete, the following builds are not done:\n",
+		h.bucket.Iteration.Fingerprint))
+	for _, b := range rem {
+		buf.WriteString(fmt.Sprintf("* %q: %s\n", b.ComponentType, b.Status))
+	}
+	buf.WriteString("You may resume work on this iteration in further Packer builds by defining the following variable in your environment:\n")
+	buf.WriteString(fmt.Sprintf("HCP_PACKER_BUILD_FINGERPRINT=%q", h.bucket.Iteration.Fingerprint))
+
+	ui.Say(buf.String())
```

Comment:
Can you have this inside the iteration object? This way you won't have it duplicated in the hcl and json registries.

Reply:
Good idea yeah, at first I implemented all that without passing in the `Ui` object, and it made a little bit more sense not to have prints from further in the chain, but now we might as well propagate this into the iteration and let it print its diags there.